dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjection/DependencyInjectionTests.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.VectorData;
+using Xunit;
+
+namespace VectorDataSpecificationTests.DependencyInjection;
+
+public abstract class DependencyInjectionTests<TKey, TRecord>
+    where TKey : notnull
+    where TRecord : notnull
+{
+    protected abstract void PopulateConfiguration(ConfigurationManager configuration, object? serviceKey = null);
+
+    protected abstract void RegisterVectorStore(IServiceCollection services, ServiceLifetime lifetime, object? serviceKey = null);
```

Comment:
Since there are several registration methods, maybe better to allow connectors to expose a list of delegates on their fixtures and perform tests over all of them.